Source/WebCore/platform/graphics/Path.h

```diff
@@ -22,426 +22,102 @@
-        return *m_type;
-    }
-
-private:
-    friend struct IPC::ArgumentCoder<EncodedPathElementType, void>;
-    std::optional<PathElement::Type> m_type;
-};
-
 class Path {
     WTF_MAKE_FAST_ALLOCATED;
 public:
     WEBCORE_EXPORT Path();
-#if USE(CG)
-    Path(RetainPtr<CGMutablePathRef>&&);
-#endif
-#if USE(CAIRO)
-    explicit Path(RefPtr<cairo_t>&&);
-#endif
-    WEBCORE_EXPORT ~Path();
+    WEBCORE_EXPORT Path(Vector<PathSegment>&&);
+    explicit Path(const Vector<FloatPoint>& points);
+    Path(std::unique_ptr<PathImpl>&&);
 
     WEBCORE_EXPORT Path(const Path&);
-    WEBCORE_EXPORT Path(Path&&);
+    Path(Path&&) = default;
     WEBCORE_EXPORT Path& operator=(const Path&);
-    WEBCORE_EXPORT Path& operator=(Path&&);
+    Path& operator=(Path&&) = default;
 
-#if ENABLE(INLINE_PATH_DATA)
-    static Path from(const InlinePathData& inlineData)
-    {
-        Path path;
-        path.m_inlineData = inlineData;
-        return path;
-    }
-#endif
-
-    static Path polygonPathFromPoints(const Vector<FloatPoint>&);
-
-    bool contains(const FloatPoint&, WindRule = WindRule::NonZero) const;
-    bool strokeContains(const FloatPoint&, const Function<void(GraphicsContext&)>& strokeStyleApplier) const;
-
-    // fastBoundingRect() should equal or contain boundingRect(); boundingRect()
-    // should perfectly bound the points within the path.
-    FloatRect boundingRect() const;
-    WEBCORE_EXPORT FloatRect fastBoundingRect() const;
-    FloatRect strokeBoundingRect(const Function<void(GraphicsContext&)>& strokeStyleApplier = { }) const;
-
-    WEBCORE_EXPORT size_t elementCount() const;
-    float length() const;
-    PathTraversalState traversalStateAtLength(float length) const;
-    FloatPoint pointAtLength(float length) const;
-
-    WEBCORE_EXPORT void clear();
-    WEBCORE_EXPORT bool isNull() const;
-    bool isEmpty() const;
-    // Gets the current point of the current path, which is conceptually the final point reached by the path so far.
-    // Note the Path can be empty (isEmpty() == true) and still have a current point.
-    // FIXME: The above comment might need to be updated; on all supported platforms, the result of hasCurrentPoint() is identical
-    // to !isEmpty().
-    bool hasCurrentPoint() const;
-    FloatPoint currentPoint() const;
-
-    bool isClosed() const;
+    WEBCORE_EXPORT bool operator==(const Path&) const;
 
     WEBCORE_EXPORT void moveTo(const FloatPoint&);
+
     WEBCORE_EXPORT void addLineTo(const FloatPoint&);
     WEBCORE_EXPORT void addQuadCurveTo(const FloatPoint& controlPoint, const FloatPoint& endPoint);
-    WEBCORE_EXPORT void addBezierCurveTo(const FloatPoint& controlPoint1, const FloatPoint& controlPoint2, const FloatPoint& endPoint);
-    void addArcTo(const FloatPoint&, const FloatPoint&, float radius);
-    WEBCORE_EXPORT void closeSubpath();
+    void addBezierCurveTo(const FloatPoint& controlPoint1, const FloatPoint& controlPoint2, const FloatPoint& endPoint);
```

Comment:
Yes you are right. But there are two naming schemes for the Path APIs: the interface CanvasPath.idl and framework header file CGPath.h. I think we mixed these two schemes and came up with this interface. Maybe we need to stick solely with one of them. Anyway let's address this in a separate patch since changing the names will touch many other areas of the code.